src/polymer-ide.ts

```diff
@@ -61,21 +68,142 @@ class PolymerIde {
       this.linter.editorService = this.editorService;
       this.autocompleter.editorService = this.editorService;
     }
-  }
+  };
 
   serialize(): ViewState {
     return {};
   };
 
   provideLinter(): lint.Provider|lint.Provider[] {
     return this.linter;
-  }
+  };
 
   provideAutocompleter(): autocomplete.Provider|autocomplete.Provider[] {
     return this.autocompleter;
-  }
+  };
 };
 
+class TooltipManager {
+  textEditor: AtomCore.IEditor;
+  tooltipMarker: AtomCore.IDisplayBufferMarker;
+  tooltipElement: HTMLElement = null;
+  documentationElement: HTMLElement = null;
+  editorService: RemoteEditorService;
+  editorElements: Array<HTMLElement> = [];
+
+  constructor(editorService: RemoteEditorService) {
+    this.editorService = editorService;
+  };
+
+  subscribe(subscriptions: CompositeDisposable) {
+    subscriptions.add(atom.workspace.observePanes((pane: AtomCore.IPane) => {
+      this.update();
+      subscriptions.add(pane.onDidChangeActiveItem(() => {
+        this.update();
+      }));
+    }));
+  };
+
+  update() {
+    this.removeTooltip();
+    this.textEditor = atom.workspace.getActiveTextEditor();
+    this.addMouseEventListener();
+  }
+
+  addMouseEventListener() {
+    const textEditorElement = atom.views.getView(this.textEditor);
+    if (!textEditorElement) {
+      return;
+    }
+    if (this.editorElements.indexOf(textEditorElement) > -1) {
+      return;
+    }
+    this.editorElements.push(textEditorElement);
+    let timer: number;
+    textEditorElement.addEventListener('mousemove', (event: AtomMouseEvent) => {
+      window.clearTimeout(timer);
+      this.removeTooltip();
```

Comment:
I think we only want to remove the tooltip if the mouse is not currently over the tooltip, and it has moved away from the tooltip by a certain amount.

This is fairly common code to write though, I wonder if we can crib from any other atom plugins for this, or if atom itself will help.